samples/mixed/consumer/src/main.rs

```diff
@@ -0,0 +1,283 @@
+use proto::digitaltwin::FindByIdRequest;
+use proto::provider::provider_client::ProviderClient;
+use proto::provider::{InvokeRequest, SubscribeRequest};
+use std::net::SocketAddr;
+use tokio::time::{sleep, Duration};
+use tonic::transport::Server;
+use uuid::Uuid;
+
+/// The ids commands.
+const ACTIVATE_AIR_CONDITIOING_COMMAND_ID: &str =
+    "dtmi:org:eclipse:sdv:vehicle:cabin:hvac:activate_air_conditioning;1";
+const SEND_NOTIFICATION_COMMAND_ID: &str =
+    "dtmi:org:eclipse:sdv:vehicle:cabin:hvac:send_notification;1";
+const SET_UI_MESSAGE_COMMAND_ID: &str = "dtmi:org:eclipse:sdv:vehicle:cabin:hvac:set_ui_message;1";
+
+/// The ids for the properties.
+const AMBIENT_AIR_TEMPERATURE_PROPERTY_ID: &str =
+    "dtmi:org:eclipse:sdv:vehicle:cabin:hvac:ambient_air_temperature;1";
+const IS_AIR_CONDITIONING_ACTIVE: &str =
+    "dtmi:org:eclipse:sdv:vehicle:cabin:hvac:is_air_conditioning_active;1";
+const HYBRID_BATTERY_REMAINING: &str =
+    "dtmi:org:eclipse:sdv:vehcile:obd:hybrid_battery_remaining;1";
+
+/// The id for the URI property.
+const URI_PROPERTY_ID: &str = "dtmi:sdv:property:uri;1";
+
+/// Start the send notification repeater.
+///
+/// # Arguments
+/// `provider_uri` - The provider_uri.
+/// `consumer_uri` - The consumer_uri.
+fn start_send_notification_repeater(provider_uri: String, consumer_uri: String) {
+    info!("Starting the Consumer's send notification repeater.");
+    tokio::spawn(async move {
+        loop {
+            // info!("Invoking the send_notification command on endpoint {}", &provider_uri);
+
+            let client_result = ProviderClient::connect(provider_uri.clone()).await;
+            if client_result.is_err() {
+                continue;
+            }
+            let mut client = client_result.unwrap();
+
+            let response_id = Uuid::new_v4().to_string();
+
+            let payload: String = String::from("The send_notification request.");
+
+            let request = tonic::Request::new(InvokeRequest {
+                entity_id: String::from(SEND_NOTIFICATION_COMMAND_ID),
+                consumer_uri: consumer_uri.clone(),
+                response_id,
+                payload,
+            });
+
+            let _response = client.invoke(request).await;
+
+            sleep(Duration::from_secs(5)).await;
+        }
+    });
+}
+
+/// Start the set ui message repeater.
+///
+/// # Arguments
+/// `provider_uri` - The provider_uri.
+/// `consumer_uri` - The consumer_uri.
+fn start_set_ui_message_repeater(provider_uri: String, consumer_uri: String) {
+    info!("Starting the Consumer's send notification repeater.");
+    tokio::spawn(async move {
+        let array: [&str; 10] =
+            ["one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"];
+        let mut index: usize = 0;
+        loop {
+            // info!("Invoking the set_ui_message command on endpoint {}", &provider_uri);
+
+            let client_result = ProviderClient::connect(provider_uri.clone()).await;
```

Comment:
same as above - do you need to reconnect every time?
Is it ok to swallow errors?